third_party/validator/sql_database_instance.go

```diff
@@ -0,0 +1,230 @@
+// ----------------------------------------------------------------------------
+//
+//     This file is copied here by Magic Modules. The code for building up a
+//     sql database instance object is copied from the manually implemented
+//     provider file:
+//     third_party/terraform/resources/resource_sql_database_instance.go.erb.go
+//
+// ----------------------------------------------------------------------------
+package google
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/googleapi"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+func GetSQLDatabaseInstanceCaiObject(d TerraformResourceData, config *Config) (Asset, error) {
+	if obj, err := GetSQLDatabaseInstanceApiObject(d, config); err == nil {
+		return Asset{
+			Name: fmt.Sprintf("//cloudsql.googleapis.com/%s", obj["selfLink"]),
+			Type: "google.cloud.sql.Instance",
+			Resource: &AssetResource{
+				Version:              "v1beta4",
```

Comment:
I'm not sure how you are using this version but it makes me nervous to see version strings hard coded. If this is a property of the sql instance, what happens if you get a sql_instance object created with a different version of the api?

What happens when the go library/api is bumped to a new version?

Reply:
I completely agree on the versioning concern. Currently this pkg is importing `google.golang.org/api/sqladmin/v1beta4`. I didnt see any version exported from that pkg (https://godoc.org/google.golang.org/api/sqladmin/v1beta4).